NPE on sessionMixin

```
Caused by: java.lang.NullPointerException: 
  at com.owncloud.android.ui.activity.BaseActivity.getStorageManager (BaseActivity.java:167)
  at com.owncloud.android.ui.fragment.FileDetailSharingFragment.onCreate (FileDetailSharingFragment.java:192)
  at androidx.fragment.app.Fragment.performCreate (Fragment.java:2586)
  at androidx.fragment.app.FragmentManagerImpl.moveToState (FragmentManagerImpl.java:838)
  at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManagerImpl.java:1238)
  at androidx.fragment.app.FragmentManagerImpl.moveToState (FragmentManagerImpl.java:1303)
  at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange (FragmentManagerImpl.java:2659)
  at androidx.fragment.app.FragmentManagerImpl.dispatchCreate (FragmentManagerImpl.java:2607)
  at androidx.fragment.app.FragmentController.dispatchCreate (FragmentController.java:235)
  at androidx.fragment.app.FragmentActivity.onCreate (FragmentActivity.java:316)
  at androidx.appcompat.app.AppCompatActivity.onCreate (AppCompatActivity.java:115)
  at com.owncloud.android.ui.activity.BaseActivity.onCreate (BaseActivity.java:57)
  at com.owncloud.android.ui.activity.ToolbarActivity.onCreate (ToolbarActivity.java:57)
  at com.owncloud.android.ui.activity.DrawerActivity.onCreate (DrawerActivity.java:1202)
  at com.owncloud.android.ui.activity.FileActivity.onCreate (FileActivity.java:168)
  at com.owncloud.android.ui.activity.FileDisplayActivity.onCreate (FileDisplayActivity.java:240)
  at android.app.Activity.performCreate (Activity.java:7955)
  at android.app.Activity.performCreate (Activity.java:7944)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1307)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3423)
```

@ezaquarii this happens on 3.11 RC1.
Do you have an idea?
From stack trace I think it might be when switching to FileDetailSharingFragment either for first time, or after suspending the app.

Yes, the fragment onCreate is running before activity on create finishes.

We must move the fragment onCreate code to onActivityCreated which is guaranteed to be called after activity onCreate fully executed.

On March 9, 2020 7:33:27 AM UTC, Tobias Kaminsky <notifications@github.com> wrote:
>```
>Caused by: java.lang.NullPointerException: 
>at com.owncloud.android.ui.activity.BaseActivity.getStorageManager
>(BaseActivity.java:167)
>at com.owncloud.android.ui.fragment.FileDetailSharingFragment.onCreate
>(FileDetailSharingFragment.java:192)
>  at androidx.fragment.app.Fragment.performCreate (Fragment.java:2586)
>at androidx.fragment.app.FragmentManagerImpl.moveToState
>(FragmentManagerImpl.java:838)
>at
>androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState
>(FragmentManagerImpl.java:1238)
>at androidx.fragment.app.FragmentManagerImpl.moveToState
>(FragmentManagerImpl.java:1303)
>at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange
>(FragmentManagerImpl.java:2659)
>at androidx.fragment.app.FragmentManagerImpl.dispatchCreate
>(FragmentManagerImpl.java:2607)
>at androidx.fragment.app.FragmentController.dispatchCreate
>(FragmentController.java:235)
>at androidx.fragment.app.FragmentActivity.onCreate
>(FragmentActivity.java:316)
>at androidx.appcompat.app.AppCompatActivity.onCreate
>(AppCompatActivity.java:115)
>at com.owncloud.android.ui.activity.BaseActivity.onCreate
>(BaseActivity.java:57)
>at com.owncloud.android.ui.activity.ToolbarActivity.onCreate
>(ToolbarActivity.java:57)
>at com.owncloud.android.ui.activity.DrawerActivity.onCreate
>(DrawerActivity.java:1202)
>at com.owncloud.android.ui.activity.FileActivity.onCreate
>(FileActivity.java:168)
>at com.owncloud.android.ui.activity.FileDisplayActivity.onCreate
>(FileDisplayActivity.java:240)
>  at android.app.Activity.performCreate (Activity.java:7955)
>  at android.app.Activity.performCreate (Activity.java:7944)
>at android.app.Instrumentation.callActivityOnCreate
>(Instrumentation.java:1307)
>at android.app.ActivityThread.performLaunchActivity
>(ActivityThread.java:3423)
>```
>
>@ezaquarii this happens on 3.11 RC1.
>Do you have an idea?
>From stack trace I think it might be when switching to
>FileDetailSharingFragment either for first time, or after suspending
>the app.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/nextcloud/android/issues/5604

--- a/src/main/java/com/owncloud/android/ui/fragment/FileDetailSharingFragment.java
+++ b/src/main/java/com/owncloud/android/ui/fragment/FileDetailSharingFragment.java
@@ -187,9 +187,6 @@ public class FileDetailSharingFragment extends Fragment implements ShareeListAda
         if (fileDisplayActivity == null) {
             throw new IllegalArgumentException("FileActivity may not be null");
         }
-
-        fileOperationsHelper = fileDisplayActivity.getFileOperationsHelper();
-        fileDataStorageManager = fileDisplayActivity.getStorageManager();
     }
 
     @Override
@@ -205,9 +202,8 @@ public class FileDetailSharingFragment extends Fragment implements ShareeListAda
         View view = inflater.inflate(R.layout.file_details_sharing_fragment, container, false);
         unbinder = ButterKnife.bind(this, view);
 
-        if (fileDataStorageManager == null) {
-            fileDataStorageManager = new FileDataStorageManager(account, fileDisplayActivity.getContentResolver());
-        }
+        fileOperationsHelper = fileDisplayActivity.getFileOperationsHelper();
+        fileDataStorageManager = fileDisplayActivity.getStorageManager();
 
         setupView();
 
@@ -233,7 +229,7 @@ public class FileDetailSharingFragment extends Fragment implements ShareeListAda
     }
 
     @Override
-    public void onAttach(Context context) {
+    public void onAttach(@NonNull Context context) {
         super.onAttach(context);
         if (!(getActivity() instanceof FileActivity)) {
             throw new IllegalArgumentException("Calling activity must be of type FileActivity");

